docker/Dockerfile.mips64el-unknown-linux-muslabi64

```diff
@@ -12,21 +12,29 @@ RUN /xargo.sh
 COPY qemu.sh /
 RUN /qemu.sh mips64el
 
+# this is a hard-float target for the mips64r2 architecture
+# https://github.com/rust-lang/rust/blob/75d3027fb5ce1af6712e4503c9574802212101bd/compiler/rustc_target/src/spec/mips64el_unknown_linux_muslabi64.rs#L6
 COPY musl.sh /
 RUN /musl.sh \
-    TARGET=mips64el-linux-muslsf \
-    "COMMON_CONFIG += -with-arch=mips64"
+    TARGET=mips64el-linux-musl \
+    "COMMON_CONFIG += -with-arch=mips64r2"
 
-ENV CROSS_MUSL_SYSROOT=/usr/local/mips64el-linux-muslsf
+ENV CROSS_MUSL_SYSROOT=/usr/local/mips64el-linux-musl
 COPY musl-symlink.sh /
-RUN /musl-symlink.sh $CROSS_MUSL_SYSROOT mips64el-sf
+RUN /musl-symlink.sh $CROSS_MUSL_SYSROOT mips64el
+RUN mkdir -p $CROSS_MUSL_SYSROOT/usr/lib64
+# needed for the C/C++ runners
+RUN ln -s $CROSS_MUSL_SYSROOT/usr/lib/libc.so $CROSS_MUSL_SYSROOT/usr/lib64/libc.so
+RUN ln -s $CROSS_MUSL_SYSROOT/usr/lib/libc.so.1 $CROSS_MUSL_SYSROOT/usr/lib64/libc.so.1
+
+COPY mips64el-linux-musl-gcc.sh /usr/bin/
 
 COPY qemu-runner base-runner.sh /
 
-ENV CARGO_TARGET_MIPS64EL_UNKNOWN_LINUX_MUSLABI64_LINKER=mips64el-linux-muslsf-gcc \
+ENV CARGO_TARGET_MIPS64EL_UNKNOWN_LINUX_MUSLABI64_LINKER=mips64el-linux-musl-gcc.sh \
     CARGO_TARGET_MIPS64EL_UNKNOWN_LINUX_MUSLABI64_RUNNER="/qemu-runner mips64el" \
-    CC_mips64el_unknown_linux_muslabi64=mips64el-linux-muslsf-gcc \
-    CXX_mips64el_unknown_linux_muslabi64=mips64el-linux-muslsf-g++ \
+    CC_mips64el_unknown_linux_muslabi64=mips64el-linux-musl-gcc \
+    CXX_mips64el_unknown_linux_muslabi64=mips64el-linux-musl-g++ \
```

Comment:
ping @abcfy2 

does this change work for you?
You can build the image with

```
cargo xtask build-docker-image mips64el-unknown-linux-muslabi64
```

this is a hard-float target according to rustc

Reply:
It should be hard-float, we're just using a soft float toolchain which is bound to cause problems at some point. A quick test is:

```bash
$ cargo init --bin hello
$ cross build --target mips64el-unknown-linux-muslabi64
$ file target/mips64el-unknown-linux-muslabi64/debug/hello
target/mips64el-unknown-linux-muslabi64/debug/hello: ELF 64-bit LSB executable, MIPS, MIPS64 rel2 version 1 (SYSV), statically linked, with debug_info, not stripped
$ readelf -A target/mips64el-unknown-linux-muslabi64/debug/hello | grep Tag_GNU_MIPS_ABI_FP
  Tag_GNU_MIPS_ABI_FP: Hard float (double precision)
$ docker run -it --rm ghcr.io/cross-rs/mips64el-unknown-linux-muslabi64:main bash
# still using the soft-float toolchain
$ echo $CROSS_MUSL_SYSROOT/
/usr/local/mips64el-linux-muslsf/

# show 32-bit is soft-float
$ cross build --target mipsel-unknown-linux-musl
$ file target/mipsel-unknown-linux-musl/debug/hello
target/mipsel-unknown-linux-musl/debug/hello: ELF 32-bit LSB pie executable, MIPS, MIPS32 rel2 version 1 (SYSV), dynamically linked, interpreter /lib/ld-musl-mipsel-sf.so.1, with debug_info, not stripped
$ readelf -A target/mipsel-unknown-linux-musl/debug/hello | grep Tag_GNU_MIPS_ABI_FP
  Tag_GNU_MIPS_ABI_FP: Soft float
```

So we're linking a hard-float executable with a soft-float toolchain, which isn't exactly ideal, and bound to cause problems later down the line.